src/app/api/projects/repo/route.ts

```diff
@@ -0,0 +1,32 @@
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { options } from "../../auth/[...nextauth]/options";
+import { registerProjectRepos } from "../projectsService";
+
+export type GitReposRegisterRequest = {
+  repos: RegisterGitRepoDto[];
+  projectId: string;
+};
+
+export type RegisterGitRepoDto = {
+  name: string;
+  full_name: string;
+};
+
+export async function POST(req: NextRequest) {
+  try {
+    const session = await getServerSession(options);
+    const registerRepoDto = (await req.json()) as GitReposRegisterRequest;
+    if (session?.userId) {
```

Comment:
check for the registerRepoDto on this condition as well, otherwise it will fail with a nullPointer